### PR TITLE
docs: Update Ci integrations config validator doc with `validate` command

### PR DESCRIPTION
**Pull request by solid-yevheniiashchypeleva**
Fixes #

## Checklist

- [x] Documented

### CODE REVIEW

File: metrics/ci_integrations/docs/04_ci_integrations_config_validator.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 ## TL;DR
 
-Introducing a `CI Integrations Config Validator` provides an ability to validate the CI Integrations configuration fields before the `sync` command to provide additional context about the possible errors in the configuration file.   
-For example, if the configuration file contains a non-valid email/password used to log in into CI, the user sees the corresponding error before the `sync` process starts.
+Introducing a `CI Integrations Config Validator` provides an ability to validate the CI Integrations configuration fields by running the `validate` command that provides additional context about the possible errors in the configuration file.
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/04_ci_integrations_config_validator.md:5]:
```suggestion
Introducing a `CI Integrations Config Validator` provides an ability to validate the CI Integrations configuration fields by running the `validate` command that provides information about the possible errors in the configuration file.   
```

File: metrics/ci_integrations/docs/04_ci_integrations_config_validator.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 ## TL;DR
 
-Introducing a `CI Integrations Config Validator` provides an ability to validate the CI Integrations configuration fields before the `sync` command to provide additional context about the possible errors in the configuration file.   
-For example, if the configuration file contains a non-valid email/password used to log in into CI, the user sees the corresponding error before the `sync` process starts.
+Introducing a `CI Integrations Config Validator` provides an ability to validate the CI Integrations configuration fields by running the `validate` command that provides additional context about the possible errors in the configuration file.   
+For example, if the configuration file contains a non-valid email/password used to log in into CI, the user sees the corresponding error before he or she will run the `sync` command.
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/04_ci_integrations_config_validator.md:6]:
```suggestion
For example, if the configuration file contains a non-valid email/password used to log in into CI, the user will get the corresponding error message once run the `validate` command.
```

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_class_diagram.puml [hunk 3]
```diff
@@ -79,10 +79,21 @@
         }
     }
 
-     package client.cool_integration as client_cool_integration {
+    package client.cool_integration as client_cool_integration {
         class CoolIntegrationClient {}
     }
 
+    package cli.command {
+        class ValidateCommand {
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_class_diagram.puml:87]: Would it be great to specify that this command extends the `CiIntegrationCommand` abstract class?

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_class_diagram.puml [hunk 4]
```diff
@@ -95,13 +106,16 @@
...
+    CoolIntegrationDestinationValidationDelegate -left-> CoolIntegrationClient : uses
     CoolIntegrationConfigValidatorFactory ..|> ConfigValidatorFactory
     CoolIntegrationConfigValidatorFactory --> CoolIntegrationDestinationConfigValidator : creates
-    CoolIntegrationConfigValidatorFactory --> CoolIntegrationClient : creates
+    CoolIntegrationConfigValidatorFactory -left-> CoolIntegrationClient : creates
     CoolIntegrationConfigValidatorFactory --> CoolIntegrationDestinationValidationDelegate : creates
     CoolIntegrationDestinationConfigValidator --> ConfigValidationError : throws
 
+    ValidateCommand -left-> IntegrationParty : uses
+    ValidateCommand -left-> ConfigValidatorFactory : uses
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_class_diagram.puml:119]: Does the `validate` command uses the `ConfigValidator` abstract class to validate the config? Also, it seems like this class uses the `Config` interface.

File: metrics/ci_integrations/docs/04_ci_integrations_config_validator.md [hunk 5]
```diff
@@ -30,16 +32,20 @@ Consider the following steps needed to be able to validate the given configurati
 1. Create the following abstract classes: `ConfigValidator`, `ValidationDelegate`, `SourceValidationDelegate`, `DestinationValidationDelegate` and `ConfigValidatorFactory`.
 2. For each source or destination party, implement its specific `ConfigValidator`, `ValidationDelegate`, and `ConfigValidatorFactory`. Implement the validation-required methods in the integration-specific clients.
 3. Add the `configValidatorFactory` to the `IntegrationParty` abstract class and provide its implementers with their party-specific config validator factories.
-4. Create the source and the destination config validators and call them within the `sync` command.
+4. Create the `ValidateCommand` class.
+5. Create the source and the destination config validators and call them within the `validate` command.
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/04_ci_integrations_config_validator.md:36]:
```suggestion
4. Create the `ValidateCommand` class.
5. Registrate a `ValidateCommand` in the `CiIntegrationsRunner`. 
6. Create the source and the destination config validators and call them within the `validate` command.
```
What do you think?

File: metrics/ci_integrations/docs/04_ci_integrations_config_validator.md [hunk 6]
```diff
@@ -85,9 +91,19 @@ Consider the package structure using the `CoolIntegration` as an example:
...
 > How will the project be tested?
 
 The project will be unit-tested using the Dart's core [test](https://pub.dev/packages/test) and [mockito](https://pub.dev/packages/mockito) packages. Also, the approaches discussed in [3rd-party API testing](https://github.com/platform-platform/monorepo/blob/master/docs/03_third_party_api_testing.md) and [here](https://github.com/platform-platform/monorepo/blob/master/docs/04_mock_server.md) should be used testing a validation client that performs direct HTTP calls. 
+
+# Alternatives Considered
+> Summarize alternative designs (pros & cons)
+
+- Implement the config validation functionality as a flag of the `sync` command (`sync --[no]-validate`).
+    - Pros:
+        - Can be enabled by default to decrease the number of errors related to invalid config files.
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/04_ci_integrations_config_validator.md:106]:
```suggestion
        - Can be enabled by default to detect invalid configuration fields before synchronization.
```

File: metrics/ci_integrations/docs/04_ci_integrations_config_validator.md [hunk 7]
```diff
@@ -85,9 +91,19 @@ Consider the package structure using the `CoolIntegration` as an example:
...
+    - Cons:
+        - No ability to validate just a config file without performing synchronization.
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/04_ci_integrations_config_validator.md:108]:
```suggestion
        - No ability to validate a config file without performing synchronization. 
```
What do you think?

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml [hunk 8]
```diff
@@ -7,33 +7,33 @@ actor Bob
...
 participant CiIntegrationsRunner
-participant SyncCommand
+participant ValidateCommand
 participant CoolIntegrationConfigValidatorFactory
 participant CoolIntegrationConfigValidator
 participant CoolIntegrationValidationDelegate
 participant CoolIntegrationClient
 
-Bob -> "CI Integrations" : runs sync
+Bob -> "CI Integrations" : runs ""validate""
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml:16]:
```suggestion
Bob -> "CI Integrations" : runs validate
```
?

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml [hunk 9]
```diff
@@ -67,11 +66,11 @@ else config is not valid
 
     CoolIntegrationConfigValidator -> CoolIntegrationConfigValidator : adds an error message to the error buffer
 
-    CoolIntegrationConfigValidator --> SyncCommand : throw ""ConfigValidationError(message)""
+    CoolIntegrationConfigValidator --> ValidateCommand : throw ""ConfigValidationError(message)""
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml:69]:
```suggestion
    CoolIntegrationConfigValidator --> ValidateCommand : throw ConfigValidationError(message)
```
?

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml [hunk 10]
```diff
@@ -67,11 +66,11 @@ else config is not valid
...
     deactivate CoolIntegrationConfigValidator
 
-    SyncCommand --> Bob : sync failed
-    deactivate SyncCommand
+    ValidateCommand --> Bob : config is not valid
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml:72]:
```suggestion
    ValidateCommand --> Bob : prints the validation error message
```

File: metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml [hunk 11]
```diff
@@ -53,9 +53,8 @@ alt config is valid
     CoolIntegrationClient --> CoolIntegrationValidationDelegate : InteractionResult.success
     CoolIntegrationValidationDelegate --> CoolIntegrationConfigValidator : InteractionResult.success
 
-    CoolIntegrationConfigValidator --> SyncCommand : config is valid
-    SyncCommand -> SyncCommand : continues the sync command
-    SyncCommand --> Bob : sync was successful
+    CoolIntegrationConfigValidator --> ValidateCommand : config is valid
+    ValidateCommand --> Bob : config is valid
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml:57]:
```suggestion
    ValidateCommand --> Bob : prints the validation successful message
```

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a couple of suggestions here - please take a look. Overall, the updated design looks clear and informative, thanks!

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-yevheniiashchypeleva & @solid-daniyilyevtyushkin!
